Sources/NIO/Selector.swift

```diff
@@ -180,83 +255,39 @@ final class Selector<R: Registration> {
+    private func register_kqueue<S: Selectable>(selectable: S, interested: SelectorEventSet, oldInterested: SelectorEventSet?) throws {
+        let oldKQueueFilters = oldInterested.map { KQueueEventFilterSet(selectorEventSet: $0) } ?? KQueueEventFilterSet._none
+        let newKQueueFilters = KQueueEventFilterSet(selectorEventSet: interested)
+        let differences = oldKQueueFilters.symmetricDifference(newKQueueFilters) // contains all the events that need a change (either need to be added or removed)
+        assert(interested.contains(.reset))
+        assert(oldInterested?.contains(.reset) ?? true)
 
-            events.0.ident = UInt(fd)
-            events.0.filter = Int16(EVFILT_READ)
-            events.0.fflags = 0
-            events.0.data = 0
-            events.0.udata = nil
-
-            events.1.ident = UInt(fd)
-            events.1.filter = Int16(EVFILT_WRITE)
-            events.1.fflags = 0
-            events.1.data = 0
-            events.1.udata = nil
-        }
-
-        switch interested {
-        case .read:
-            events.0.flags = UInt16(EV_ADD)
-            events.1.flags = UInt16(EV_DELETE)
-        case .write:
-            events.0.flags = UInt16(EV_DELETE)
-            events.1.flags = UInt16(EV_ADD)
-        case .all:
-            events.0.flags = UInt16(EV_ADD)
-            events.1.flags = UInt16(EV_ADD)
-        case .none:
-            events.0.flags = UInt16(EV_DELETE)
-            events.1.flags = UInt16(EV_DELETE)
-        }
-
-        var offset: Int = 0
-        var numEvents: Int32 = 2
-
-        if let old = oldInterested {
-            switch old {
-            case .read:
-                if events.1.flags == UInt16(EV_DELETE) {
-                    numEvents -= 1
-                }
-            case .write:
-                if events.0.flags == UInt16(EV_DELETE) {
-                    offset += 1
-                    numEvents -= 1
-                }
-            case .none:
-                // Only discard the delete events
-                if events.0.flags == UInt16(EV_DELETE) {
-                    offset += 1
-                    numEvents -= 1
-                }
-                if events.1.flags == UInt16(EV_DELETE) {
-                    numEvents -= 1
-                }
-            case .all:
-                // No need to adjust anything
-                break
-            }
-        } else {
-            // If its not reregister operation we MUST NOT include EV_DELETE as otherwise kevent will fail with ENOENT.
-            if events.0.flags == UInt16(EV_DELETE) {
-                offset += 1
-                numEvents -= 1
-            }
-            if events.1.flags == UInt16(EV_DELETE) {
-                numEvents -= 1
+        func calculateKQueueChange(event: KQueueEventFilterSet) -> UInt16? {
+            guard differences.contains(event) else {
+                return nil
             }
+            return UInt16(newKQueueFilters.contains(event) ? EV_ADD : EV_DELETE)
```

Comment:
Similarly, should this be a method on `KQueueEventFilterSet`?

Reply:
same here